src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -101,55 +106,77 @@
 
       (munge-fn version url-prefix))))
 
-(defn stream-query-result
+(def query-options-schema
```

Comment:
produce-streaming-body and stream-query-results now takes an options hash that gets validated and has defaults. We keep moving around these parameters and this feels like a step in a better direction to manage these, at least produce-streaming-body has a contract with pe-puppetdb-extensions for example.